content/en/docs/quarks-operator/concepts/ha.md

```diff
@@ -109,20 +111,47 @@ Currently:
 > True if this instance is the first instance of its group.
 ([BOSH jobs](https://bosh.io/docs/jobs/))
 
-The pod that starts first in a statefulet must have the bootstrap flag. It's used to initialize databases and such.
+The pod that starts first in a statefulset must have the bootstrap flag. It's used to initialize databases and such.
 
 In case of an upgrade, the last pod, for example pod-3 has the bootstrap flag. If the pod is restarted, the bootstrap flag has to stay the same.
 
-The startup-ordinal is used to determine this flag.
+The startup-ordinal must be used to determine this flag.
+
+### Calculation
+
+The startup-ordinal is converted into the startup-index:
+If AZs are not used (AZ<2) it's equal to the startup-ordinal, otherwise it's  `(azIndex-1) * 10000 + startup-ordinal`.
+
+
+Now an array of all possible job instances is generated.
+
+It has `bootstrapIndex = ig.Instances*azCount - 1`
+
+* `Instances` is the number of instances of the job.
+* `azCount` is always at least 1.
+
+So for a job in 2 AZs, with 3 instances: 5
+
+Unless someone sets initial rollout, then it's always 0.
+
+*`initial_rollout`* is true, only if the template-rendering container was created for the ig resolved manifest with v1.
+*`bootstrap`* is true, when the `instance.index` is equal to the bootstrap index.
+*`instance.index` is sequential, starting at 0, across azs. (example: 0..5), becomes the spec.index
+
+This array of job instances is created for each job in the instance group and added to ig struct (`ig.Jobs[].Properties.Quarks.Instances=[]jobInstances`).
+
+(Pre-Render Scripts run for the instance group.)
+
+Then, with the pre-generated arrays, for each job, render the jobs templates. Pick the pre-generated job instance where the instance.index matches the startup-index.
+Unless AZs are used, then  the startup-index needs to match `instance.Instance+azIndex*10000`.
 
 ## spec.index
 
 > Instance index. Use spec.bootstrap to determine the first instead of checking whether the index is 0. Additionally, there is no guarantee that instances will be numbered consecutively, so that there are no gaps between different indices.
 ([BOSH jobs](https://bosh.io/docs/jobs/))
 
-The label value seems to be unsused and the index is calculated instead: If AZs are not used (AZ<2) it's equal to the startup-ordinal, otherwise it's 10000 + startup-ordinal.
+The label value seems to be unsused and the index is calculated instead. It's the sequential number of the jobs instance, across AZs.
```

Comment:
As above, I'm hoping that the "sequential number of the jobs instance" means `(azIndex-1)*ig.Instances + podOrdinal`; please make this explicit, if true.